### ISSUE TITLE
Using strategy.spinnaker.io/replace along with strategy.spinnaker.io/max-version-history makes replace not work

### ISSUE THREAD

**Issue by billiford** [homedepot/go-clouddriver#168]
Deploy two config maps, one with just the `strategy.spinnaker.io/replace` annotation and another with both annotations `strategy.spinnaker.io/replace` and `strategy.spinnaker.io/max-version-history`. Note that only the first removes the annotation `kubectl.kubernetes.io/last-applied-configuration` generated by kubectl on replace. The second config map still has this annotation, which for large config maps results in this error on replace:
```
error applying manifest (kind: ConfigMap, apiVersion: v1, name: test-v000): ConfigMap "test-v000" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```
![image](https://github.com/homedepot/go-clouddriver/assets/7597848/c31c0e42-3500-406d-a067-99dd34c1ecaf)

**Comment by abe21412** [comment]
it looks like when the `max-version-history` annotation (or really any numerical annotation) is added without explicit quotes on the value it gets parsed as an int when converted to JSON. So when the request makes it to go-clouddriver, any time [`Unstructured.GetAnnotations()`](https://github.com/kubernetes/apimachinery/blob/60eaa653342bd0cc48626fa2604bb6d3e7171706/pkg/apis/meta/v1/unstructured/unstructured.go#L412) is called, the below error occurs internally but gets ignored because that func doesn't examine the return values of [`NestedStringMap()`](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/helpers.go#L166). This causes the annotations to be reset to empty when go-clouddriver begins adding Spinnaker's reserved annotations.

`.metadata.annotations accessor error: contains non-string key in the map: 10 is of the type int64, expected string`

**Comment by billiford** [comment]
Excellent find, thank you!

**Comment by billiford** [comment]
@abe21412, @guido9j, shouldn't we close this issue since https://github.com/homedepot/go-clouddriver/pull/175 fixed it?

**Comment by billimek** [comment]
Fixed by #175
